### PR TITLE
Add ci task to publish api client

### PR DESCRIPTION
**Pull request by crobibero**
Uses https://openapi-generator.tech/ to generate api clients.
Not sure how to hook up to npmjs, and I probably missed something else.

Should publish unstable to azure repo, stable to npmjs.

### CODE REVIEW

File: .ci/azure-pipelines-api-client.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    - task: DownloadPipelineArtifact@2
+      displayName: 'Download OpenAPI Spec Artifact'
+      inputs:
+        source: "specific"
+        artifact: "OpenAPI Spec"
+        path: "$(System.ArtifactsDirectory)/openapi"
+        project: "$(System.TeamProjectId)"
+        pipeline: "29" # The main server CI build
+        runVersion: "latestFromBranch"
+        runBranch: "refs/heads/$(System.PullRequest.TargetBranch)"
```
Comment (EraYaN) [inline .ci/azure-pipelines-api-client.yml:26]: If you want (and add the `dependsOn: Test` to this job, you can download from the current build. That might be more in line with what would be expected. (See the OpenAPI Spec SSH upload job for an example)

**Review by EraYaN — APPROVED** [review]
Apart from the openapi.json download it looks good.

File: .ci/azure-pipelines-api-client.yml (hunk 1 above)
Comment (EraYaN) [reply .ci/azure-pipelines-api-client.yml:26]: Something like 
```yaml
  - task: DownloadPipelineArtifact@2
    displayName: 'Download OpenAPI Spec'
    inputs:
      source: 'current'
      artifact: "OpenAPI Spec"
      path: "$(System.ArtifactsDirectory)/openapispec"
      runVersion: "latest"
```

From the PR #4249

File: apiclient/templates/typescript/axios/unstable.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+java -jar openapi-generator-cli.jar generate \
+    --input-spec $(System.ArtifactsDirectory)/openapi/openapi.json \
+    --generator-name typescript-axios \
+    --output ./apiclient/generated/typescript/axios  \
+    --template-dir ./apiclient/templates/typescript/axios \
+    --ignore-file-override ./apiclient/.openapi-generator-ignore \
+    --additional-properties=useSingleRequestParameter="true",npmName="axios",snapshotVersion="-SNAPSHOT.$(Build.BuildNumber)",npmRepository="https://dev.azure.com/jellyfin-project/jellyfin/_packaging"
```
Comment (EraYaN) [inline apiclient/templates/typescript/axios/unstable.sh:9]: The actual repo url is `https://pkgs.dev.azure.com/jellyfin-project/jellyfin/_packaging/unstable/npm/registry/ `

**Comment by EraYaN** [comment]
This also depends on https://github.com/jellyfin/jellyfin/pull/4249

**Comment by crobibero** [comment]
Partly depends on https://github.com/jellyfin/jellyfin/pull/4255 for proper docs

File: .ci/azure-pipelines-api-client.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  steps:
+    - task: DownloadPipelineArtifact@2
+      displayName: 'Download OpenAPI Spec Artifact'
+      inputs:
+        source: 'current'
+        artifact: "OpenAPI Spec"
+        path: "$(System.ArtifactsDirectory)/openapispec"
+        runVersion: "latest"
+        dependsOn: Test
```
Comment (EraYaN) [inline .ci/azure-pipelines-api-client.yml:24]: This should go on the actual job (now line 11-12)

File: .ci/azure-pipelines-api-client.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    - task: CmdLine@2
+    displayName: 'Download OpenApi Generator'
```
Comment (EraYaN) [inline .ci/azure-pipelines-api-client.yml:27]: This indentation, is that correct? Use the Azure Pipelines vs code plugin to validate the yaml. I think it needs an extra two spaces.

File: .ci/azure-pipelines.yml [hunk 5]
```diff
@@ -55,3 +55,6 @@ jobs:
 
 - ${{ if or(startsWith(variables['Build.SourceBranch'], 'refs/tags/v'), startsWith(variables['Build.SourceBranch'], 'refs/heads/master')) }}:
   - template: azure-pipelines-package.yml
+
+- ${{ if or(startsWith(variables['Build.SourceBranch'], 'refs/tags/v'), startsWith(variables['Build.SourceBranch'], 'refs/heads/master')) }}:
+    - template: azure-pipelines-api-client.yml
```
Comment (EraYaN) [inline .ci/azure-pipelines.yml:60]:
```suggestion
  - template: azure-pipelines-api-client.yml
```

File: .ci/azure-pipelines-api-client.yml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  pool:
+    vmImage: "${{ parameters.LinuxImage }}"
+
+  steps:
+    - task: DownloadPipelineArtifact@2
+      displayName: 'Download OpenAPI Spec Artifact'
+      inputs:
+        source: 'current'
+        artifact: "OpenAPI Spec"
+        path: "$(System.ArtifactsDirectory)/openapispec"
```
Comment (EraYaN) [inline .ci/azure-pipelines-api-client.yml:23]: You use `$(System.ArtifactsDirectory)/openapispec` here but `$(System.ArtifactsDirectory)/openapi` in the actual bash scripts.

File: .ci/azure-pipelines-api-client.yml [hunk 7]
```diff
@@ -0,0 +1,62 @@
+﻿parameters:
+  - name: LinuxImage
+    type: string
+    default: "ubuntu-latest"
+  - name: GeneratorVersion
+    type: string
+    default: "4.3.1"
```
Comment (Camc314) [inline .ci/azure-pipelines-api-client.yml:7]:
```suggestion
    default: "5.0.0-beta2"
```
Currenly there are issues with typescript APIs with the generator on 4.x versions (see [here](https://github.com/OpenAPITools/openapi-generator/issues/7282)). This is fixed in the v5 beta so targeting the v5 beta should fix the issue.

**Review by joshuaboniface — APPROVED** [review]

**Pull request merged**